pkgs/applications/graphics/rapid-photo-downloader/default.nix

```diff
@@ -0,0 +1,77 @@
+{ stdenv, fetchurl, python3Packages
+, file, intltool, gobjectIntrospection, libgudev
+, udisks, glib, gnome3, gst_all_1, libnotify
+, exiv2, exiftool, qt5, gdk_pixbuf
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "rapid-photo-downloader";
+  version = "0.9.9";
+
+  src = fetchurl {
+    url = "https://launchpad.net/rapid/pyqt/${version}/+download/${pname}-${version}.tar.gz";
+    sha256 = "193l55wdxwxmammyxwzp90sg0qw5r316443la3m9611sdpcmy8w3";
+  };
+
+  # Disable version check and fix install tests
+  postPatch = ''
+    substituteInPlace raphodo/constants.py \
+      --replace "disable_version_check = False" "disable_version_check = True"
+    substituteInPlace raphodo/rescan.py \
+      --replace "from preferences" "from raphodo.preferences"
+    substituteInPlace raphodo/copyfiles.py \
+      --replace "import problemnotification" "import raphodo.problemnotification"
+  '';
+
+  nativeBuildInputs = [ file intltool gobjectIntrospection ];
+
+  buildInputs = [
+    # Fix "ValueError: Namespace GUdev not available" by building typelib files
+    # for GUdev. Remove as soon as PR #35335 is merged, i.e. GObject
+    # Introspection has been added to libgudev's nativeBuildInputs by default.
+    (libgudev.overrideAttrs (oldAttrs: rec {
+      nativeBuildInputs = oldAttrs.nativeBuildInputs ++ [ gobjectIntrospection ];
+    }))
+
+    udisks
+    glib
+    gnome3.gexiv2
+    gst_all_1.gstreamer
+    libnotify
+    exiv2
+    exiftool
+    qt5.qtimageformats
+    gdk_pixbuf
+  ] ++ (with python3Packages; [
+    pyqt5
+    pygobject3
+    gphoto2
+    pyzmq
+    tornado
+    psutil
+    pyxdg
+    arrow
+    dateutil
+    easygui
+    colour
+    pymediainfo
+    sortedcontainers
+    rawkit
+    requests
+    colorlog
+    pyprind
+  ]);
+
+  makeWrapperArgs = [
+    "--set GI_TYPELIB_PATH \"$GI_TYPELIB_PATH\""
+    "--set PYTHONPATH \"$PYTHONPATH\""
```

Comment:
this will add build-time dependencies as run-time dependencies. Is the default wrapper not sufficient?

Reply:
This was the only way I was able to make it work.

Reply:
Makes sense, because the Python packages you've added are not included in `propagatedBuildInputs`. We use that attribute also for indicating they're run-time dependencies.

Reply:
I've tried that before but then it can't find some modules like `zmq`.